.github/workflows/docker_publish.yaml

```diff
@@ -35,10 +29,10 @@ jobs:
           images: |
             unleashorg/unleash-server
           tags: |
-            # only enabled for v* tags:
-            type=semver,pattern={{ version }},enable=${{ startsWith(github.ref, 'refs/tags/v') }}
-            type=semver,pattern={{ major.minor }},enable=${{ startsWith(github.ref, 'refs/tags/v') }}
-            type=semver,pattern={{ major }},enable=${{ startsWith(github.ref, 'refs/tags/v') }}
+            # only enabled for workflow dispatch except main (assume its a release):
+            type=semver,pattern={{ version }},enable=${{ github.event_name == 'workflow_dispatch' && github.ref != 'refs/heads/main' }}
+            type=semver,pattern={{ major.minor }},enable=${{ github.event_name == 'workflow_dispatch' && github.ref != 'refs/heads/main' }}
+            type=semver,pattern={{ major }},enable=${{ github.event_name == 'workflow_dispatch' && github.ref != 'refs/heads/main' }}
```

Comment:
We can't check that the ref starts with tags because the ref will be the one of the branch we're releasing from. Instead, whenever this workflow is called will tag the image with semver, except when called from main (cause we're not releasing from main)